packages/dev/core/src/Engines/WebGPU/Extensions/engine.computeShader.ts

```diff
@@ -93,6 +101,11 @@ WebGPUEngine.prototype.computeDispatch = function (
         computePass.dispatchWorkgroups(x, y, z);
     }
     computePass.end();
+
+    if (frameTimeObject) {
+        this._timestampQuery.endPass(this._timestampIndex, frameTimeObject);
+        this._timestampIndex += 2;

```

Comment:
Why +2?

Reply:
Because there are two indices used internally, one for the `startPass` timestamp, and one for the `endPass` timestamp (the duration being calculated as `end_timestamp - start_timestamp`).